pkg/migration/azure_crs.go

```diff
@@ -391,6 +396,158 @@ func (m *azureMigrator) readRelease(ctx context.Context, ver string) error {
 	return nil
 }
 
+func (m *azureMigrator) updateCluster(ctx context.Context) error {
+	cluster := m.crs.cluster
+
+	// Drop operator version label.
+	delete(cluster.Labels, label.AzureOperatorVersion)
+
+	// Drop finalizers.
+	cluster.Finalizers = nil
+
+	// Adjust k8s apiserver bind port to match kubeadm.
+	if cluster.Spec.ClusterNetwork != nil && cluster.Spec.ClusterNetwork.APIServerPort != nil {
+		cluster.Spec.ClusterNetwork.APIServerPort = to.Int32Ptr(6443)
+	}
+
+	cluster.Spec.ControlPlaneRef = &corev1.ObjectReference{
+		APIVersion: m.crs.kubeadmControlPlane.APIVersion,
+		Kind:       m.crs.kubeadmControlPlane.Kind,
+		Name:       m.crs.kubeadmControlPlane.Name,
+	}
+
+	err := m.mcCtrlClient.Update(ctx, cluster)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func (m *azureMigrator) updateAzureCluster(ctx context.Context) error {
+	cluster := m.crs.azureCluster
+
+	// Drop operator version label.
+	delete(cluster.Labels, label.AzureOperatorVersion)
+
+	// Drop finalizers.
+	cluster.Finalizers = nil
+
+	// Use default credentials.
+	cluster.Spec.IdentityRef = nil
```

Comment:
Can you explain that line?

Reply:
Whoops, that's leftover from my earlier testing. We must not nullify that anymore. Earlier when I tested by moving to separate cluster, the credentials were pre-configured there so resetting that simplified the testing setup & migration code when I focused on solving the etcd migration problem. I'll drop that.